client/js/components/GameView/MoveChoiceOverlay.vue

```diff
@@ -70,181 +70,173 @@ export default {
     },
   },
   computed: {
-    /**
-     * Returns list of objects representing the available moves,
-     * based on the selected card
-     */
-    moveChoices() {
-      if (!this.selectedCard) return [];
-      let res = [];
-      let cardRank;
-      if (this.resolvingSeven) {
-        if (!this.cardSelectedFromDeck) return [];
-        cardRank = this.cardSelectedFromDeck.rank;
-      } else {
-        if (!this.selectedCard) return [];
-        cardRank = this.selectedCard.rank;
-      }
-      // Re-usable move-objects
-      const allMovesAreDisabled = !this.isPlayersTurn || this.frozenId === this.selectedCard.id;
-      const allMovesDisabledExplanation = !this.isPlayersTurn
-        ? "It's not your turn"
-        : 'This card is frozen';
-
-      let disabledText;
-      if (allMovesAreDisabled) {
-        disabledText = allMovesDisabledExplanation;
+    // Determines if any moves are available
+    allMovesAreDisabled() {
+      return !this.isPlayersTurn || this.frozenId === this.selectedCard.id;
+    },
+    // Determines which disabled text to display
+    disabledText() {
+      if (this.allMovesAreDisabled) {
+        return !this.isPlayersTurn ? "It's not your turn" : 'This card is frozen';
       }
-
-      // Points
-      const pointsDescription = `Gain ${cardRank} point` + (cardRank === 1 ? '' : 's');
-      const pointsMove = {
+      return '';
+    },
+    pointsMove() {
+      if (!this.selectedCard) return [];
+      const pointsDescription =
+        `Gain ${this.selectedCard.rank} point` + (this.selectedCard.rank === 1 ? '' : 's');
+      return {
         displayName: 'Points',
         eventName: 'points',
         moveDescription: pointsDescription,
-        disabled: allMovesAreDisabled,
-        disabledExplanation: disabledText,
+        disabled: this.allMovesAreDisabled,
+        disabledExplanation: this.disabledText,
       };
-      // Scuttling
+    },
+    scuttleMove() {
+      if (!this.selectedCard) return [];
```

Comment:
Same here
```suggestion
      if (!this.selectedCard) return null;
```

Reply:
per our conversation, let's actually make `selectedCard` a required prop and remove these null checks instead